sensor/kubernetes/listener/resource_event_handler.go

```diff
@@ -76,9 +80,22 @@ func (k *listenerImpl) handleAllEvents() {
 	var crdSharedInformerFactory dynamicinformer.DynamicSharedInformerFactory
 	var complianceResultInformer, complianceProfileInformer, complianceTailoredProfileInformer, complianceScanSettingBindingsInformer, complianceRuleInformer, complianceScanInformer, complianceSuiteInformer cache.SharedIndexInformer
 	var profileLister cache.GenericLister
-	if resourceList, err := serverResourcesForGroup(k.client, complianceoperator.GetGroupVersion().String()); err != nil {
-		log.Errorf("Checking API resources for group %q: %v", complianceoperator.GetGroupVersion().String(), err)
-	} else if resourceExists(resourceList, complianceoperator.ComplianceCheckResult.Name) {
+	crdWatcher := crd.NewCRDWatcher(&k.stopSig, dynamicinformer.NewDynamicSharedInformerFactory(k.client.Dynamic(), noResyncPeriod))
+	coAvailabilityChecker := complianceOperatorAvailabilityChecker.NewComplianceOperatorAvailabilityChecker()
+	if err := coAvailabilityChecker.AppendToCRDWatcher(crdWatcher); err != nil {
+		log.Errorf("Unable to add the Resource to the CRD Watcher: %v", err)
+	}
+	if err := crdWatcher.Watch(k.crdWatcherStatusC); err != nil {
+		log.Errorf("failed to start watching the CRDs: %v", err)
+	}
+	crdHandlerFn := func(status *watcher.Status) {
+		if status.Available {
+			log.Infof("Resources %v became available", status.Resources)
+		} else {
+			log.Infof("Resources %v became unavailable", status.Resources)
```

Comment:
Is this `else` case ever called? IIUC the logic, `crdHandlerFn` will be called if the CRDs change. If they are going from unavailable to available, then this message will be shown `Resource %v became available`, and sensor will soft restart -- recreating this listeners. That means that seeing it going from available to unavailable can only happen if the CRDs were **available** during listeners creation.